execute/plugin_functions.go

```diff
@@ -249,15 +248,20 @@ func mergeMessageObservations(
 
 	results := make(exectypes.MessageObservations)
 	for selector, validator := range validators {
-		msgs := validator.GetValid()
-		if _, ok := results[selector]; !ok {
-			results[selector] = make(map[cciptypes.SeqNum]cciptypes.Message)
-		}
-		for _, msg := range msgs {
-			results[selector][msg.Header.SequenceNumber] = msg
+		if msgs := validator.GetValid(); len(msgs) > 0 {
+			if _, ok := results[selector]; !ok {
+				results[selector] = make(map[cciptypes.SeqNum]cciptypes.Message)
+			}
+			for _, msg := range msgs {
+				results[selector][msg.Header.SequenceNumber] = msg
+			}
 		}
 	}
 
+	if len(results) == 0 {
+		return nil, nil
+	}
```

Comment:
Added extra logic to these merge functions to make sure `nil` is returned instead of empty data structures. This makes the test cases slightly simpler.